koku/masu/external/kafka_msg_handler.py

```diff
@@ -237,6 +238,22 @@ def construct_parquet_reports(request_id, context, report_meta, payload_destinat
     return daily_parquet_files
 
 
+def remove_invalid_csv(dir_path):
+    for (root, _, filenames) in os.walk(dir_path):
+        csv_reports = [fl for fl in filenames if fl.endswith(".csv")]
+        for csv_report in csv_reports:
+            file_path = f"{root}/{csv_report}"
+            try:
+                pd.read_csv(file_path)
+            except Exception as error:
+                LOG.warning(f"File {file_path} is not a valid CSV file. Reason: {str(error)}")
+                LOG.debug(f"Removing file {file_path} because it is invalid CSV")
+                try:
+                    os.remove(file_path)
```

Comment:
Instead of removing the file can we just log enough information so that we know what account the payload came from.  Then `extract_payload` would return `None` so that no further processing is done.

We might have to examine what's in the Kafka message to figure out if we can determine the account the payload came from.

Reply:
I thought we needed to remove the invalid files so that processing of valid files could continue?

Reply:
@maskarb what do you think about processing partially correct payloads?  I'm sort of on the fence.  On hand hand, if we have some reports (pod, storage, etc) that are good it would be good to just process it.  On the other if one is bad, failing the entire payload would bring it to the user and therefore our attention must faster.